homeassistant/components/tuya/config_flow.py

```diff
@@ -121,12 +120,13 @@ async def async_step_login(self, user_input=None):
             )
 
             if response.get("success", False):
-                _LOGGER.debug("TuyaConfigFlow.async_step_user login success")
+                _LOGGER.debug("Login success: %s", response)
                 return self.async_create_entry(
                     title=user_input[CONF_USERNAME],
                     data=user_input,
                 )
             errors["base"] = RESULT_AUTH_FAILED
+            _LOGGER.error("Login failed: %s", response)
```

Comment:
we generally only log unexpected errors. If we are showing it to user, there shouldn't be a reason to log it anything but debug. Once this current issue is resolved, there is no reason to have failed login attempts be logged like this?

Reply:
That makes for a terrible user experience...
But I guess that's fine.

Reply:
Errors can [reference description placeholders](https://github.com/home-assistant/frontend/blob/dev/src/dialogs/config-flow/show-dialog-config-flow.ts#L94-L99). That way we can inject error codes.